File: biblioteca_app/lib/services/auth_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:biblioteca_app/config/api_config.dart'; // Importa tu configuración dinámica

class AuthService {
  Future<Map<String, dynamic>> loginRequest(String correo, String password) async {
    // Usamos ApiConfig.login en lugar de la variable estática
    final response = await http.post(
      Uri.parse(ApiConfig.login),
      headers: {
        'Content-Type': 'application/json',
      },
      body: jsonEncode({
        'correo': correo, 
        'password': password,
      }),
    );

    if (response.statusCode != 200) {
      final error = jsonDecode(response.body);
      throw Exception(error['error'] ?? 'Error al iniciar sesión');
    }

    return jsonDecode(response.body);
  }
}

File: biblioteca-inf/src/services/authService.js
const API_URL = 'http://localhost:3000/api/auth';

export const loginRequest = async (data) => {
    const response = await fetch(`${API_URL}/login`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(data),
    });

    if (!response.ok) {
        const error = await response.json();
        throw new Error(error.error);
    }

    return await response.json();
};